### ISSUE TITLE
Import and export exposure notification data 

### ISSUE THREAD

**Issue by rugk** [microg/android_packages_apps_GmsCore#1208]
## Use case

I have used Google Play Services before.
Now I want to switch to microG.
Because of the Corona/COVID-19-situation I do not want to loose my data.

## How to

What I can currently do:
Go to the COVID screen:
![corona-google](https://user-images.githubusercontent.com/11966684/95024810-c9251600-0685-11eb-9d4f-e9f2e1e5d439.png)

Export the data:
![corona-google-export](https://user-images.githubusercontent.com/11966684/95024819-d215e780-0685-11eb-8c16-b3f03f280f50.png)

**Problem:** After you press that menu entry, you get literally **no** feedback. I have **no idea** where that data is saved. (checking the usual internal storage I could not find anything.)

## Intended solution

Anyway, what we need in microG's implementation is just an import feature for those.
For whatever reason, Google Play Services do not offer any import feature though – maybe so one cannot abuse it and generate fake results. (though AFAIK this is already cryptographically assured, as you cannot easily generate those hashes or respectively only random ones hmm)
Anyway, would be great if we could switch to microG but keeping the 14days history, so one does not loose data that could result in a missed COVID notification.

For completeness you may of course also implement an export feature – I guess if you have an import feature that will be easy. So you'd have feature parity with Google's implementation in this regard then.

**Comment by mar-v-in** [comment]
@rugk do you have any details on the export format? Does it actually contain the history of collected ids or just the history of risk calculations? From the description it sounds like the latter.

**Comment by rugk** [comment]
> do you have any details on the export format?

No, because as said I could not even find the file/where it exports those. :see_no_evil: (Any help appreciated… :see_no_evil: )

**Comment by rugk** [comment]
BTW of course a silly, but technically possible solution to this problem is also: Have another phone with you with Google Play Services for >= 14days (or it may be possible to only do that at the end of the 14 days) and then upgrade your main phone to microG. Afterwards, keep your other phone for 14 days too and then you can also upgrade the other phone to microG. 
If the other phone did not get any notification in these 14 days then it's fine, hehe…

Of course, this is a little weird and not everyone has two phones at their disposal, so this is only a workaround,

**Comment by mh-** [comment]
Sorry to say that, but it’s only a report, not the actual RPI data. 
To get the actual RPI (and your own TEK) data, you need to have access to the GMS files on your device, i.e. root access. 
They are stored in LevelDBs, see e.g. here for sample code: https://github.com/mh-/corona-warn-companion-android/blob/8e2dc84998a73440eccbc6028e0cc9d02fc2a2ad/corona-warn-companion/src/main/java/org/tosl/coronawarncompanion/gmsreadout/ContactDbOnDisk.java#L52

**Comment by rugk** [comment]
> Sorry to say that, but it’s only a report, not the actual RPI data.

Ok, anyway, and how can one get this report?
Also note you can click on the time it shows you in the second screen and then it shows you some hash and stuff, I had guessed this is all you need.

**Comment by mh-** [comment]
The purpose of the report is: you can check what your government’s app is doing. If they e.g. sent different DK files to you than to other people, you could see different hashes, and you could notice that they give you a special treatment...

On my device (Android 9), the report itself seems to be available for any type of export I want. E.g.: send by e-mail

**Comment by rugk** [comment]
> On my device (Android 9), the report itself seems to be available for any type of export I want. E.g.: send by e-mail

Ah, so it should open the "share menu" of Android? Hmm… does not seem to happen/work in my case. :thinking:

**Comment by rugk** [comment]
Feature request [on Mastodon](https://bunt.social/@maximpistos/105281090431592727) for this. Seedvault unfortunately does not include microG data in the backup…

**Comment by tlambertz** [comment]
I have just filed a pull request which allows you to export the database from microg (#1291). You could then check the keys with the corona companion app, once [corona-warn-companion#97](https://github.com/mh-/corona-warn-companion-android/pull/97) is implemented. This is not exactly the same as a re-import, but should give similar (in my opinion even better) results.

**Comment by mar-v-in** [comment]
b7717e5 adds support for backups using adb backup/restore or seedvault. Backups will also cover exposure.db.
